packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormRawJsonFieldInput.tsx

```diff
@@ -0,0 +1,104 @@
+import { FormFieldInputContainer } from '@/object-record/record-field/form-types/components/FormFieldInputContainer';
+import { FormFieldInputInputContainer } from '@/object-record/record-field/form-types/components/FormFieldInputInputContainer';
+import { FormFieldInputRowContainer } from '@/object-record/record-field/form-types/components/FormFieldInputRowContainer';
+import { TextVariableEditor } from '@/object-record/record-field/form-types/components/TextVariableEditor';
+import { useTextVariableEditor } from '@/object-record/record-field/form-types/hooks/useTextVariableEditor';
+import { VariablePickerComponent } from '@/object-record/record-field/form-types/types/VariablePickerComponent';
+import { InputLabel } from '@/ui/input/components/InputLabel';
+import { CAPTURE_VARIABLE_TAG_REGEX } from '@/workflow/search-variables/utils/initializeEditorContent';
+import { useId } from 'react';
+import { isDefined } from 'twenty-ui';
+import { turnIntoEmptyStringIfWhitespacesOnly } from '~/utils/string/turnIntoEmptyStringIfWhitespacesOnly';
+
+type FormRawJsonFieldInputProps = {
+  label?: string;
+  defaultValue: string | null | undefined;
+  placeholder: string;
+  onPersist: (value: string | null) => void;
+  readonly?: boolean;
+  VariablePicker?: VariablePickerComponent;
+};
+
+export const FormRawJsonFieldInput = ({
+  label,
+  defaultValue,
+  placeholder,
+  onPersist,
+  readonly,
+  VariablePicker,
+}: FormRawJsonFieldInputProps) => {
+  const inputId = useId();
+
+  const editor = useTextVariableEditor({
+    placeholder,
+    multiline: true,
+    readonly,
+    defaultValue: defaultValue ?? undefined,
+    onUpdate: (editor) => {
+      const text = turnIntoEmptyStringIfWhitespacesOnly(editor.getText());
+
+      if (text === '') {
+        onPersist(null);
+
+        return;
+      }
+
+      const textWithReplacedVariables = text.replaceAll(
+        new RegExp(CAPTURE_VARIABLE_TAG_REGEX, 'g'),
+        '0',
+      );
+
+      let isValidJson;
+      try {
+        JSON.parse(textWithReplacedVariables);
+
+        isValidJson = true;
+      } catch {
+        isValidJson = false;
+      }
+
+      if (!isValidJson) {
+        return;
+      }
```

Comment:
I am not a big fan of this. We will want to valid the field indeed. But we do not want to avoid persisting without letting the user notice. This is why we do not have any validation on links and email yet, we leave the user save whatever he wants.

Next step would be to discuss with Thomas on a common behavior for field content validation. Also maybe not let the workflow being activated if a field is invalid so we know the workflow would fail.

I would:
- remove that step
- add a fast follow to discuss validation

Wdyt?

Reply:
I agree that we didn't talk about validation yet. In fact, I already added validation to the Date field in a way. The input doesn't allow you to submit invalid dates.
For the Raw JSON field, I can remove the validation step for now.